schema/crdb/crucible-ref-count-records/up08.sql

```diff
@@ -0,0 +1,29 @@
+INSERT INTO volume_resource_usage (
+  SELECT
+    (
+      OVERLAY(
+       OVERLAY(
+        MD5(volume.id::TEXT || dataset_id::TEXT || region_id::TEXT || snapshot_id::TEXT || snapshot_addr || volume_references::TEXT)
+        PLACING '4' from 13
```

Comment:
What is this doing? Why '4'? Why `from 13`?

Reply:
This code creates a deterministic V4 UUID from the other columns, which [has the shape of](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random))

```
xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx
12345678 9111 1111 1112 222222222333
          012 3456 7890 123456789012
```

where `the first hexadecimal digit in the third group always starts with a 4` means that M = 4, and I put a "random" value (read: volume references) in for N (the variant field). 

But from https://datatracker.ietf.org/doc/html/rfc9562#name-uuid-version-4:

> Alternatively, an implementation MAY choose to randomly generate the exact required number of bits for random_a, random_b, and random_c (122 bits total) and then concatenate the version and variant in the required position.

I don't think I'm doing this right - position 17 shouldn't be "random":

> var:
>     The 2-bit variant field as defined by [Section 4.1](https://datatracker.ietf.org/doc/html/rfc9562#variant_field), set to 0b10.

I can fix this tomorrow

Reply:
done in 58169feb6 (plus a comment)

Reply:
I am leaving a comment so that Github stops bringing me to this text box. Please ignore!